From hackermail Thu Jan  1 00:00:00 1970
From: SeongJae Park <sj@kernel.org>
To: Andrew Morton <akpm@linux-foundation.org>
Cc: SeongJae Park <sj@kernel.org>, Shuah Khan <shuah@kernel.org>, damon@lists.linux.dev, kernel-team@meta.com, linux-kernel@vger.kernel.org, linux-kselftest@vger.kernel.org, linux-mm@kvack.org
Message-Id: <20250628160428.53115-3-sj@kernel.org>
In-Reply-To: <20250628160428.53115-1-sj@kernel.org>
Date: Sat, 28 Jun 2025 09:04:24 -0700
Subject: [PATCH 2/6] selftests/damon/_damon_sysfs: set Kdamond.pid in start()

_damon_sysfs.py is a Python module for reading and writing DAMON sysfs
for testing.  It is not reading resulting kdamond pids.  Read and update
those when starting kdamonds.

Signed-off-by: SeongJae Park <sj@kernel.org>
Link: https://patch.msgid.link/20250628160428.53115-3-sj@kernel.org
Cc: Shuah Khan <shuah@kernel.org>
---
 tools/testing/selftests/damon/_damon_sysfs.py | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/tools/testing/selftests/damon/_damon_sysfs.py b/tools/testing/selftests/damon/_damon_sysfs.py
index 5b1cb6b3ce4e..f587e117472e 100644
--- a/tools/testing/selftests/damon/_damon_sysfs.py
+++ b/tools/testing/selftests/damon/_damon_sysfs.py
@@ -408,6 +408,9 @@ class Kdamond:
             if err is not None:
                 return err
         err = write_file(os.path.join(self.sysfs_dir(), 'state'), 'on')
+        if err is not None:
+            return err
+        self.pid, err = read_file(os.path.join(self.sysfs_dir(), 'pid'))
         return err
 
     def stop(self):
-- 
2.39.5